cli/src/lockfiles/mod.rs

```diff
@@ -24,10 +24,7 @@ pub trait Parse {
     fn parse(&self, data: &str) -> ParseResult;
 
     /// Parse from a file
-    fn parse_file<P: AsRef<Path>>(&self, path: P) -> ParseResult
-    where
-        Self: Sized,
-    {
+    fn parse_file(&self, path: &Path) -> ParseResult {
```

Comment:
Changing this forces all of the callers to use `Path::new()` just so that one of the callers (`extensions/api.rs`) doesn't have to call `std::fs::read_to_string()`...

I think it would have made much more sense to have the extension code read the file and then call `parse(data)` instead of `parse_file(path)`.

Reply:
I agree that having the generics looks infinitely better; the problem is not `parse_file`, per se, but the fact that a trait object can not be instantiated in the first place if one of the trait methods have a bound on `Self: Sized`.

A trait object is by definition `!Sized` and a method `where Self: Sized` cannot exist at all, hence the trait cannot be fully implemented on a trait object. Unfortunately, we do need to construct a trait object as a consequence of moving away from closures for the parsers.

Reply:
Right. But this function is just a convenience function around `Parse::parse()`, which has no `Sized` requirement and can be called on trait objects.

So any code that needs to use trait objects can and should just call `Parse::parse()`

Reply:
Indeed, for some reason I believed you couldn't create trait objects _at all_ with sized-constrained trait methods, but I have now verified I was wrong. Reverted this change in the latest commit.